File: cpp/周赛/279/1.cpp


class Solution {
public:
    vector<int> sortEvenOdd(vector<int> &nums) {
        vector<int> n1;
        vector<int> n2;
        for (int i = 0; i < nums.size(); i++) {
            if (i % 2 == 1) {
                n1.push_back(nums[i]);
            } else {
                n2.push_back(nums[i]);
            }
        }
        sort(n1.begin(), n1.end(), [](int a, int b) { return a > b; });
        sort(n2.begin(), n2.end());
        for (int i = 0; i < nums.size(); i++) {
            if (i % 2 == 1) {
                nums[i] = n1[i / 2];
            } else {
                nums[i] = n2[i / 2];
            }
        }
        return nums;
    }
};

int main() {

}

File: python/0 competition/单/288/1.py
# Author: Jintao Huang
# Email: hjt_study@qq.com
# Date:
from typing import List

class Solution:
    def largestInteger(self, num: int) -> int:
        nums = [int(x) for x in str(num)]
        even = [x for x in nums if x % 2 == 0]
        odd = [x for x in nums if x % 2 == 1]
        even.sort(key=lambda x:-x)
        odd.sort(key=lambda x:-x)
        j, k = 0, 0
        for i in range(len(nums)):
            x = nums[i]
            if x % 2 == 0:
                nums[i] = even[j]
                j += 1
            else:
                nums[i] = odd[k]
                k += 1
        return int("".join(str(x) for x in nums))


print(Solution().largestInteger(1234))
print(Solution().largestInteger(65875))